beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -494,71 +493,48 @@ pub enum PeerConnectionStatus {
-impl Serialize for PeerConnectionStatus {
-    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let mut s = serializer.serialize_struct("connection_status", 6)?;
-        match self {
-            Connected { n_in, n_out } => {
-                s.serialize_field("status", "connected")?;
-                s.serialize_field("connections_in", n_in)?;
-                s.serialize_field("connections_out", n_out)?;
-                s.serialize_field("last_seen", &0)?;
-                s.end()
-            }
-            Disconnecting { .. } => {
-                s.serialize_field("status", "disconnecting")?;
-                s.serialize_field("connections_in", &0)?;
-                s.serialize_field("connections_out", &0)?;
-                s.serialize_field("last_seen", &0)?;
-                s.end()
-            }
-            Disconnected { since } => {
-                s.serialize_field("status", "disconnected")?;
-                s.serialize_field("connections_in", &0)?;
-                s.serialize_field("connections_out", &0)?;
-                s.serialize_field("last_seen", &since.elapsed().as_secs())?;
-                s.serialize_field("banned_ips", &Vec::<IpAddr>::new())?;
-                s.end()
-            }
-            Banned { since } => {
-                s.serialize_field("status", "banned")?;
-                s.serialize_field("connections_in", &0)?;
-                s.serialize_field("connections_out", &0)?;
-                s.serialize_field("last_seen", &since.elapsed().as_secs())?;
-                s.end()
-            }
-            Dialing { since } => {
-                s.serialize_field("status", "dialing")?;
-                s.serialize_field("connections_in", &0)?;
-                s.serialize_field("connections_out", &0)?;
-                s.serialize_field("last_seen", &since.elapsed().as_secs())?;
-                s.end()
-            }
-            Unknown => {
-                s.serialize_field("status", "unknown")?;
-                s.serialize_field("connections_in", &0)?;
-                s.serialize_field("connections_out", &0)?;
-                s.serialize_field("last_seen", &0)?;
-                s.end()
-            }
-        }
+mod instant {
+    use serde::de::Error;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+    use std::time::{Duration, Instant};
+
+    pub fn serialize<S>(instant: &Instant, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let duration = instant.elapsed();
+        duration.serialize(serializer)
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Instant, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let duration = Duration::deserialize(deserializer)?;
+        let now = Instant::now();
+        let instant = now
+            .checked_sub(duration)
+            .ok_or_else(|| Error::custom("checked_sub error"))?;
```

Comment:
I think we need to avoid erroring in this case, or it'll create issues when translating `Instant`s between machines. If the duration we get off the wire is greater than our current instant then it'll error. Ideally we could use `saturating_sub`, but it doesn't exist for `Instant`, so I think our best option is probably to do `.checked_sub(duration).unwrap_or_else(Instant::now)`.

Reply:
Un-addressed comment by author @leruaa. Double checking there's still demand to call lighthouse only APIs programatically with the internal eth2 tooling?

Reply:
I think we can close this, as Jack has implemented it more recently in https://github.com/sigp/lighthouse/pull/4937